react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

```diff
@@ -579,33 +586,42 @@ public void toggleEditorMode(boolean htmlModeEnabled) {
     }
 
     public void appendUploadMediaFile(final int mediaId, final String mediaUri, final boolean isVideo) {
-       if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
-           String mediaType = getMediaType(isVideo);
-           mMediaPickedByUserOnBlock = false;
-           List<RNMedia> mediaList = new ArrayList<>();
-           mediaList.add(new Media(mediaId, mediaUri, mediaType));
-           mPendingMediaUploadCallback.onUploadMediaFileSelected(mediaList);
-       } else {
-           // we can assume we're being passed a new image from share intent as there was no selectMedia callback
-           sendOrDeferAppendMediaSignal(mediaId, mediaUri, isVideo);
-       }
+//       if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
+//           String mediaType = getMediaType(isVideo);
+//           mMediaPickedByUserOnBlock = false;
+//           List<RNMedia> mediaList = new ArrayList<>();
+//           mediaList.add(new Media(mediaId, mediaUri, mediaType));
+//           mPendingMediaUploadCallback.onUploadMediaFileSelected(mediaList);
+//       } else {
+//           // we can assume we're being passed a new image from share intent as there was no selectMedia callback
+//           sendOrDeferAppendMediaSignal(mediaId, mediaUri, isVideo);
+//       }
     }
 
     public void appendUploadMediaFiles(ArrayList<Media> mediaList) {
         if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
             mMediaPickedByUserOnBlock = false;
             List<RNMedia> rnMediaList = new ArrayList<>();
-            for (Media media : mediaList) {
-                rnMediaList.add(new Media(media.getId(), media.getUrl(), media.getType()));
+
+            if (mAppendsMultipleSelectedToSiblingBlocks && 1 < mediaList.size()) {
```

Comment:
Good suggestion! I've added a clarifying comment, and also a note about when in the future this may no longer be needed (after image block <-> gallery block transforms are implemented on mobile).